app/assets/javascripts/utilities/showUserAlertModal.js

```diff
@@ -30,24 +30,67 @@ function showUserAlertModal(title, text, confirm_text) {
  * Displays a user rate limit alert modal letting the user know what they did that exceeded a rate limit,
  * and gives them links to explain why they must wait
  *
- * @function showUserAlertModal
+ * @function showRateLimitModal
  * @param {string} action_text Description of the action taken by the user
  * @param {string} next_action_text Description of the next action that can be taken
  *
  * @example
  * showRateLimitModal('Made a comment', 'comment again')
  */
-function showRateLimitModal(action_text, next_action_text) {
-  let rateLimitText = buildRateLimitText(action_text, next_action_text);
+function showRateLimitModal(
+  action_text,
+  next_action_text,
+  timeframe = 'a moment',
+) {
+  let rateLimitText = buildRateLimitText(
+    action_text,
+    next_action_text,
+    timeframe,
+  );
   let rateLimitLink = '/faq';
   showUserAlertModal(
-    'Wait a moment...',
+    `Wait ${timeframe}...`,
     rateLimitText,
     'Got it',
     rateLimitLink,
     'Why do I have to wait?',
   );
 }
+/**
+ * Displays the corresponding modal after an error.
+ *
+ * @function showModalAfterError
+ * @param {Object} response The response from the API
+ * @param {string} action_text Description of the action taken by the user
+ * @param {string} next_action_text Description of the next action that can be taken
+ * @param {string} timeframe Description of the time that we need to wait
+ *
+ * @example
+ * showModalAfterError(response, 'made a comment', 'making another comment', 'a moment');
+ */
+function showModalAfterError(
+  response,
+  action_text,
+  next_action_text,
+  timeframe,
+) {
+  response
+    .json()
+    .then(function parseError(errorReponse) {
+      if (response.status === 429) {
+        showRateLimitModal(action_text, next_action_text, timeframe);
+      } else {
+        showUserAlertModal(
+          `Error ${next_action_text}`,
```

Comment:
(blocking) - Unfortunately I don't think `next_action_text` always makes sense in this situation, because what is being passed in assumes we are making _another_ comment/image upload/reaction. But actually a general error can happen for a wide variety of reasons - e.g. this is the response after I try to comment with the same text as I commented earlier:

![screenshot of error dialog with title "Error making another comment"](https://user-images.githubusercontent.com/20773163/131325378-ebe14c30-db42-4303-83c8-6aa91a1ac3a6.png)

In this context "another comment" doesn't make as much sense as the error we used to show, and we're also losing the additional text in the description -  ("Error posting comment" / "Your comment could not be posted due to an error: body markdown has already been taken").

The problem is the same for the catch block on line 91 - we show "Error making another comment" but a server error could happen on the first comment/reaction/etc.